lib/api/BackbeatAPI.js

```diff
@@ -601,33 +605,29 @@ class BackbeatAPI {
     }
 
     /**
-     * Delete a Redis key.
-     * @param {String} key - The key to delete
-     * @param {Function} cb - The callback function
+     * Delete all object-level metrics keys.
+     * @param {String} bucket - The bucket that the key belongs to
+     * @param {String} key - The key
+     * @param {String} versionId - The key's version ID
+     * @param {String} storageClass - The storage class of the CRR operation
+     * @param {Function} cb - The callback to call
      * @return {undefined}
      */
-    _deleteKey(key, cb) {
-        const cmd = ['del', key];
-        return this._redisClient.batch([cmd], (err, res) => {
-            if (err) {
-                this._logger.error('error deleting redis key', {
-                    method: 'BackbeatAPI._deleteKey',
-                    key,
-                    error: err,
-                });
-                return cb(err);
-            }
-            const [cmdErr] = res[0];
-            if (cmdErr) {
-                this._logger.error('error deleting redis key', {
-                    method: 'BackbeatAPI._deleteKey',
-                    key,
-                    error: cmdErr,
-                });
-                return cb(cmdErr);
-            }
-            return cb();
-        });
+    _deleteProgressKeys(bucket, key, versionId, storageClass, cb) {
+        const keys = [
+            getObjectBytesKey(storageClass, bucket, key, versionId),
+            getObjectBytesDoneKey(storageClass, bucket, key, versionId),
+        ];
+        return async.each(keys, (key, next) =>
+            this._redisClient.keys(`${key}*`, (err, res) => {
+                if (err) {
+                    return next(err);
+                }
+                return async.eachLimit(res, 10, (key, done) =>
```

Comment:
the implementation looks good. was wondering if it is possible to use `batch` for deleting the multiple keys

Reply:
👍 I agree, that would be more efficient. My one concern is the case of a big upload where there could be hundreds of interval keys to delete. I'll just confirm some performance assumptions around `batch`.